numpy/lib/function_base.py

```diff
@@ -2242,17 +2243,116 @@ def disp(mesg, device=None, linefeed=True):
     return
 
 
+# See http://docs.scipy.org/doc/numpy/reference/c-api.generalized-ufuncs.html
+_DIMENSION_NAME = r'\w+'
+_CORE_DIMENSION_LIST = '(?:%s(?:,%s)*)?' % (_DIMENSION_NAME, _DIMENSION_NAME)
```

Comment:
Sure, that's a little better. Neither is especially readable for building regular expressions, to be honest.